### ISSUE TITLE
Make ImageAnimationView more "dynamic"

### ISSUE THREAD

**Issue by jobe-m** [korlibs/korge-next#405]
I am thinking of using ImageAnimationView (or ImageDataView) for objects like explosions which are animated with a sequence of pixel art frames (aseprite). (No, I am not going to use particle effects for that ;-) It should look old school :-)

For that it would be nice if the ImageAnimationView object would play the animation only once and then stop. Currently we have there only FORWARD, REVERSE and PING_PONG enums (ImageAnimation.Direction). They all play endlessly.

So could we implement something like ONCE_FORWARD and ONCE_REVERSE. Which would stop with the last frame?

Also after the animation finished playing how could we "reuse" the ImageAnimationView for the time we need to play an explosion?
Should the object be deactivated and removed from the Container (removeChild...) and put into an object Pool? So that we do not create always a new object.
So when the animation finishes playing with "ONCE_FORWARD". It could stop the updater() and call a callback so that the parent object manager knows to deactivate the ImageAnimationView object.

Does this sound reasonable? Please make suggestions if there are better ways to implement that.
I would like to look into this topic. So you can assign this issue to me.

**Comment by soywiz** [comment]
That's an excellent idea @jobe-m !

Since you have done already several nice and significant contributions, can I add you to the github org so you can self-assign this issue yourself and use korge-next directly?

**Comment by jobe-m** [comment]
Of course you can. I am very much honoured by your move. I would nevertheless create always PRs for my contibutions.

**Comment by soywiz** [comment]
❤️

I really appreciate your contributions Marko. I have sent you the invite already.

You should be able to assign the issue to yourself as member. Also you can add the issue to the Korlibs org board and optionally adjust priority and other settings here:

<img width="352" alt="Screenshot 2021-12-17 at 08 41 50" src="https://user-images.githubusercontent.com/570848/146507945-9b27696b-701c-4413-9db2-2433b5fa349d.png">

https://github.com/orgs/korlibs/projects/19

master is blocked I guess, so probably it still needs an approval from one member to merge, but at least CI should trigger now automatically.
I just started my true holidays this week 🎄 (both OSS and this week Work) so I might be slow to respond on discord and here yet, specially reviewing PRs.

**Comment by jobe-m** [comment]
Thanks for inviting me :) Have nice and relaxing holidays!

**Comment by jobe-m** [comment]
I am still working on it :)

While thinking about a "Sprite" handling system to control sprite animations I discovered ECS Entity Component Systems the second time since I started game developing. This time I though that I give it a try to understand how an ECS works and what benefits it brings.
While trying to implement an own small ECS for my purpose and to learn more about ECS in general I realized that there is a nice implementation of a ECS in Kotlin -> https://github.com/Quillraven/Fleks
Thanks to Loganv79 :)

Unfortunately it does not work out of the box with Korge since it makes heavily use of java reflections, anotations and other stuff.
But I desided to give it a try and port it to Korge. That means to make it multiplatform compatible.
I will soon push a branch with my progress. Its going slowly but I have a start...
